packages/atomic/src/components/commerce/atomic-commerce-search-box-recent-queries/atomic-commerce-search-box-recent-queries.ts

```diff
@@ -51,37 +50,36 @@ export class AtomicCommerceSearchBoxRecentQueries {
    * - Use a value that starts with `assets://`, to display an icon from the Atomic package.
    * - Use a stringified SVG to display it directly.
    */
-  @Prop() public icon?: string;
+  @property() public icon?: string;
 
   /**
    * The maximum number of suggestions to display when the user types in the input field.
    */
-  @Prop({reflect: true}) public maxWithQuery = 3;
+  @property({type: Number, attribute: 'max-with-query', reflect: true})
+  public maxWithQuery = 3;
   /**
    * The maximum number of suggestions to display initially, when the input field is empty.
    */
-  @Prop({reflect: true}) public maxWithoutQuery?: number;
+  @property({type: Number, attribute: 'max-without-query', reflect: true})
+  public maxWithoutQuery?: number;
 
-  componentWillLoad() {
+  connectedCallback() {
+    super.connectedCallback();
     try {
-      dispatchSearchBoxSuggestionsEvent<SearchBox, Bindings>(
+      dispatchSearchBoxSuggestionsEvent<SearchBox, CommerceBindings>(
         (bindings) => {
           this.bindings = bindings;
           return this.initialize();
         },
-        this.host,
+        this,
         ['atomic-commerce-search-box']
       );
     } catch (error) {
       this.error = error as Error;
     }
   }
 
-  private renderIcon() {
-    return this.icon || Clock;
-  }
-
-  private initialize(): SearchBoxSuggestions {
+  public initialize(): SearchBoxSuggestions {
```

Comment:
is this really a component that gets  `initialized` by anything else (e.g. bind-state, mixins bindings, ...)?
If not I don't think the component should implement the `InitializableComponent` as it does not makes sense here.

Specially if this method returns some html templates... to me the name is kinda confusing